clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -410,40 +410,58 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
             }
         }
         if (shard->prev_instr_.instr.addr != 0 /*first*/) {
-            report_if_false(
-                shard, // Filtered.
+
```

Comment:
I find that these blank lines disrupt the existing grouping: see how in this file each check has a comment and a block of code with no blank lines inside the block for that check but with blank lines separating each check (see e.g. line 258 and line 278) and that's how they're grouped to help in visually seeing each check.  This particular check has gotten very large though.  It would probably be best to move this all to a helper function, or maybe turn the header comment at line 375 into a banner (i.e., whole line of *'s) and have a blank line before it and another banner line at the bottom to better group this and then we can have separators inside (the function seems the better choice).